application/apps/indexer/parsers/src/someip.rs

```diff
@@ -0,0 +1,330 @@
+use crate::{Error, LogMessage, Parser};
+use std::{fmt, fmt::Display, io::Write};
+
+use someip_messages::*;
+use someip_payload::fibex::FibexModel;
+use someip_payload::fibex2som::FibexTypes;
+use someip_payload::som::{SOMParser, SOMType};
+
+use log::debug;
+use serde::Serialize;
+
+pub struct SomeipParser<'m> {
+    model: &'m FibexModel,
+}
+
+impl<'m> SomeipParser<'m> {
+    pub fn new(model: &'m FibexModel) -> Self {
+        SomeipParser { model }
+    }
+}
+
+impl<'m> Parser<SomeipLogMessage> for SomeipParser<'m> {
+    fn parse<'a>(
+        &mut self,
+        input: &'a [u8],
+        timestamp: Option<u64>,
+    ) -> Result<(&'a [u8], Option<SomeipLogMessage>), Error> {
+        let time = timestamp.unwrap_or(0);
+        match Message::from_slice(input) {
+            Ok(Message::Sd(header, payload)) => {
+                let len = header.message_len();
+                debug!("at {} : SD Message ({:?} bytes)", time, len);
+                Ok((
+                    &[],
```

Comment:
`parse` results in an optional `SomeipLogMessage` along with the rest of the data that was handed to the parser. So returning an empty [] is not correct.
`parse` will be called with some rather big buffer that can easily contain more data then just 1 message

Reply:
Done. Raised fix for: https://github.com/esrlabs/someip/pull/12